eth/eventhandler/handlers.go

```diff
@@ -233,11 +233,21 @@ func (eh *EventHandler) handleShareCreation(
 		sharePublicKeys,
 		encryptedKeys,
 	)
+
+	var malformedEventError *MalformedEventError = nil
+
 	if err != nil {
-		return nil, fmt.Errorf("could not extract validator share from event: %w", err)
+		if errors.As(err, &malformedEventError) &&
+			malformedEventError.IsInvalidEncryptedShare &&
+			share.BelongsToOperator(eh.operatorData.GetOperatorData().ID) {
```

Comment:
this error can't ever happen if this is not your validator, because you only try to decrypt your own shares, so maybe this check is redundant?